### PR TITLE
upgrade android ndk to r17c

### PR DESCRIPTION
**Pull request by jakubgs**
Required by #7666
I've pushed the image and it's already up:
https://cloud.docker.com/u/statusteam/repository/docker/statusteam/status-build-android/tags

### CODE REVIEW

**Review by rasom — APPROVED** [review]

**Comment by jakubgs** [comment]
Doesn't seem the codebase likes the upgrade:
```
:react-native-android:buildReactNdkLib
A problem was found with the configuration of task ':react-native-android:buildReactNdkLib'. Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated and is scheduled to be removed in Gradle 5.0.
 - File '/home/jenkins/workspace/status-react_prs_android_PR-7698/node_modules/react-native/ReactAndroid/src/main/jni/react' specified for property '$1' is not a file.
/usr/lib/android-ndk/build/core/setup-app-platform.mk:135: *** Android NDK: Aborting    .  Stop.
Android NDK: APP_PLATFORM set to unknown platform: android-16.    
make[1]: Entering directory `/home/jenkins/workspace/status-react_prs_android_PR-7698/node_modules/react-native/ReactAndroid/src/main/jni/react/jni'
make[1]: Leaving directory `/home/jenkins/workspace/status-react_prs_android_PR-7698/node_modules/react-native/ReactAndroid/src/main/jni/react/jni'
:react-native-android:buildReactNdkLib FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-native-android:buildReactNdkLib'.
> Process 'command '/usr/lib/android-ndk/ndk-build'' finished with non-zero exit value 2
```
Not sure what `APP_PLATFORM` does but we set it to `android-18`:
https://github.com/status-im/status-react/blob/38aca129cd65216078381d10abb7fb7347fce427/android/app/jni/Application.mk#L1-L4

**Comment by pedropombeiro** [comment]
Wondering if we should do this after we've merged the `feature/nix` branch

**Comment by rasom** [comment]
@PombeirP why?

File: .TOOLVERSIONS [hunk 1]
```diff
@@ -1,5 +1,5 @@
-android-ndk;r10e;070be287539e3e7706f8dabfb6bf9879
-android-sdk-build-tools;28.0.1;
+android-ndk;r17c;a4b6b8281e7d101efd994d31e64af746
+android-sdk-build-tools;28.0.3;
 android-sdk-platform;android-27;
```
Comment (rasom) [inline .TOOLVERSIONS]: @PombeirP we also need to use android-28, can we change this as well? Or this config doesn't have effect on docker image?

Comment (pedropombeiro) [reply .TOOLVERSIONS]: OK, I'll push a new image with that change too. In your #status-core message you mentioned

> we need to upgrade android-sdk-build-tools to 28.0.3 and android-sdk-platform to android-27

that's why I didn't change it

**Comment by pedropombeiro** [comment]
> @PombeirP why?

Because all these dockerfiles disappear in that branch, so we'll need to redo this work in Nix before merging to develop.

File: .TOOLVERSIONS (hunk 1 above)
Comment (rasom) [reply .TOOLVERSIONS]: oh I see, that was a mistake, sorry

**Comment by rasom** [comment]
> so we'll need to redo this work in Nix before merging to develop

ok, I see. The problem here is that it probably will be a bit hard to maintain this PR because it's quite big, so if we switch to Nix soon then no problem to merge it after. Otherwise it might be a bit tricky with conflicts, already rebased it twice (though without huge difficulties).

**Comment by rasom** [comment]
> to maintain this PR because it's quite big

I mean another PR https://github.com/status-im/status-react/pull/7666 :)

File: .TOOLVERSIONS [hunk 2]
```diff
@@ -1,6 +1,6 @@
-android-ndk;r10e;070be287539e3e7706f8dabfb6bf9879
-android-sdk-build-tools;28.0.1;
-android-sdk-platform;android-27;
+android-ndk;r17c;a4b6b8281e7d101efd994d31e64af746
+android-sdk-build-tools;28.0.3;
+android-sdk-platform;android-28;
 android-sdk;4333796;aa190cfd7299cd6a1c687355bb2764e4
```
Comment (rasom) [inline .TOOLVERSIONS:4]: should this line be changed? how do we manage this file btw? is this some docker related stuff?

File: .TOOLVERSIONS [hunk 3]
```diff
@@ -1,6 +1,6 @@
-android-ndk;r10e;070be287539e3e7706f8dabfb6bf9879
-android-sdk-build-tools;28.0.1;
-android-sdk-platform;android-27;
+android-ndk;r17c;a4b6b8281e7d101efd994d31e64af746
+android-sdk-build-tools;28.0.3;
+android-sdk-platform;android-28;
```
Comment (rasom) [inline .TOOLVERSIONS:3]: btw it seems that `android-26` is hardcoded here https://github.com/status-im/status-react/blob/6d3211de46f302ab1a92eb3bb711733dd9cb8d9f/docker/android/Dockerfile#L13

Comment (pedropombeiro) [reply .TOOLVERSIONS:3]: I'm not too familiar with the Android SDK dependencies, I assume that we need this baseline 26 platform for some reason.

File: .TOOLVERSIONS (hunk 2 above)
Comment (pedropombeiro) [reply .TOOLVERSIONS:4]: This file is read when generating the Dockerfile to figure out which files to incorporate in it. It is also used to tag the Docker image with a different hash, depending on the values we put here.

Comment (rasom) [reply .TOOLVERSIONS:4]: ok, understood

File: .TOOLVERSIONS (hunk 3 above)
Comment (jakubgs) [reply .TOOLVERSIONS:3]: Yeah, it is, that's kinda how `sdkmanager` works, the packages included are named after the version of sdk. I guess it could be parametrized.

Comment (jakubgs) [reply .TOOLVERSIONS:3]: Ah wait, it actually already is parametrized right below it:
https://github.com/status-im/status-react/blob/6d3211de46f302ab1a92eb3bb711733dd9cb8d9f/docker/android/Dockerfile#L14
So my guess would be that when I was building this it wouldn't work if also `platform:android-26` was not installed.

**Comment by pedropombeiro** [comment]
When I build the app locally, I do see this:

```
Configuration 'compile' in project ':react-native-image-crop-picker' is deprecated. Use 'implementation' instead.
WARNING: The specified Android SDK Build Tools version (26.0.1) is ignored, as it is below the minimum supported version (26.0.2) for Android Gradle Plugin 3.0.1.
Android SDK Build Tools 26.0.2 will be used.
To suppress this warning, remove "buildToolsVersion '26.0.1'" from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools.
```

The `buildToolsVersion "26.0.1"` directive is in `node_modules/react-native-image-crop-picker/android./build.gradle`

Maybe we should just update that module. cc @jeluard

**Comment by jakubgs** [comment]
Considering we've merged the Nix branch and the way to manage NDK is this file now:
https://github.com/status-im/status-react/blob/develop/scripts/lib/setup/nix/mobile/android-ndk/default.nix
This PR is no obsolete.
